app/Models/AccessPoint.php

```diff
@@ -2,8 +2,40 @@
 
 namespace App\Models;
 
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use LibreNMS\Interfaces\Models\Keyable;
+
 class AccessPoint extends DeviceRelatedModel
 {
-    protected $primaryKey = 'accesspoint_id';
+    use HasFactory;
+
+    public $primaryKey = 'accesspoint_id';
     public $timestamps = false;
+
+    protected $fillable = [
+        'device_id',
+        'name',
+        'radio_number',
+        'type',
+        'mac_addr',
+        'deleted',
+        'channel',
+        'txpow',
+        'radioutil',
+        'numasoclients',
+        'nummonclients',
+        'numactbssid',
+        'nummonbssid',
+        'interference',
+    ];
+
+    public function getCompositeKey()
+    {
+        return "$this->mac_addr-$this->radio_number";
+    }
+
+    public function setOffline()
```

Comment:
https://laravel.com/docs/8.x/eloquent#soft-deleting

Reply:
With this implemented, I would just need to use withThrashed to get all aps?

Reply:
@ottorei correct

Reply:
and it stores the deleted date, so we can do a cleanup.

Reply:
> 
> 
> and it stores the deleted date, so we can do a cleanup.

Yes, with a user configurable amount of days.